core/src/test/java/org/elasticsearch/search/aggregations/bucket/HistogramIT.java

```diff
@@ -349,15 +377,19 @@ public void testSingleValuedFieldWithSubAggregation() throws Exception {
                 }
             }
             assertThat(sum.getValue(), equalTo((double) s));
-            assertThat((long) propertiesKeys[i], equalTo((long) i * interval));
-            assertThat((long) propertiesDocCounts[i], equalTo(valueCounts[i]));
-            assertThat((double) propertiesCounts[i], equalTo((double) s));
+            assertThat(propertiesKeys[i], equalTo((long) i * interval));
```

Comment:
I tend to prefer `assertEquals` over `assertThat(..., equalTo(...))` in *most* cases. This is one. I think the casts aren't become unnecessary in that case but I'm not sure. Can you check on it while you are here?